src/Dfe.PlanTech.Web/Helpers/CustomTelemetryInitializer.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Dfe.PlanTech.Web.Helpers;
+
+public class CustomTelemetryInitializer : ITelemetryInitializer
+{
+    
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public CustomTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public void Initialize(ITelemetry telemetry)
+    {
+        if (telemetry is not RequestTelemetry requestTelemetry) return;
+        var requestId = GetRequestId();
+        requestTelemetry.Context.Properties["RequestId"] = requestId;
```

Comment:
There's some warnings about this being obsolete; not sure if it will be an issue?